ecommerce/extensions/payment/views/xstack.py

```diff
@@ -0,0 +1,172 @@
+            endpoint = getattr(api_url, api_resource_name)
+            endpoint().get()
+        except Exception:  # pylint: disable=broad-except
+            logger.exception('Failed to send enrollment notification for [%s] [%s] from LMS.', user, course_key)
+
+    def _get_basket(self, payment_id):
+        """
+        Retrieve a basket using a payment ID.
+
+        Arguments:
+            payment_id: payment_id received from payment intent url generated by xstack payment processor.
+
+        Returns:
+            It will return related basket or log exception and return None if
+            duplicate payment_id received or any other exception occurred.
+
+        """
+        try:
+            basket = PaymentProcessorResponse.objects.get(
+                processor_name=self.payment_processor.NAME,
+                transaction_id=payment_id
+            ).basket
+        except MultipleObjectsReturned:
+            logger.warning(u"Duplicate payment ID [%s] received from PostEx.", payment_id)
+
+            if 'Redirection' not in self.processor_message:
+                return None
+
+            logger.info('Looking into multiple baskets for view')
+            basket = PaymentProcessorResponse.objects.filter(
+                processor_name=self.payment_processor.NAME,
+                transaction_id=payment_id
+            )[0].basket
+        except Exception:  # pylint: disable=broad-except
+            logger.exception(u"Unexpected error during basket retrieval while executing PostEx payment.")
+            return None
+
+        basket.strategy = strategy.Default()
+        basket_add_organization_attribute(basket, self.request.GET)
+        return basket
+
+    def post(self, request):
+        """
+        This creates a payment intent , manages order and basket status,
+        creates and reciept url that is then shown to user
+        """
+        secret_key = self.payment_processor.configuration['secret_key']
+        hmac_secret = self.payment_processor.configuration['hmac_secret']
+        account_id = self.payment_processor.configuration['account_id']
+        data = XStackPostBackSerializer(data=request.data)
+        data.is_valid(raise_exception=True)
+        basket_res = request.GET.dict()
+        payment_id = basket_res['orderRefNum']
+
+        payload = OrderedDict([
+            ('amount', basket_res['amount']),
+            ('currency', "PKR"),
+            ('payment_method_types', "card"),
+            ('customer', OrderedDict([
+                ('email', data.data['email']),
+                ('name', '{} {}'.format(data.data['first_name'], data.data['last_name'])),
+                ('phone', data.data['phone_number']),
+            ])),
+            ('shipping', OrderedDict([
+                ('address1', '{}, {}'.format(data.data['street_address'], data.data['address_line2'])),
+                ('city', data.data['city']),
+                ('country', ''),
```

Comment:
@hamzatanveer012 pass country and province data as passed from the updated form by Tanveer.